### PR TITLE
better Favorites

### PR DESCRIPTION
**Pull request by nstringham**
After many merging pains, I am proud to present a PR that should improve the multiple favorites system with zero merge conflicts.

### CODE REVIEW

**Comment by Pottedplant530** [comment]
@metalcupcake5 this mans about to ruin your whole carrier

**Comment by metalcupcake5** [comment]
hamsterspook

File: public/resources/js/stats.js [hunk 1]
```diff
@@ -915,105 +915,45 @@ document.addEventListener('DOMContentLoaded', function(){
...
+        if(uuid == "0c0b857f415943248f772164bf76795c"){
+            favoriteNotification.setContent("No");
+        }else{
+            let cookieArray = parseFavorites(getCookie("favorite"));
+            if(cookieArray.includes(uuid)){
+                cookieArray.splice(cookieArray.indexOf(uuid), 1);
 
-                notification.show();
+                favoriteNotification.setContent("Removed favorite!");
+            }else{
```
Comment (metalcupcake5) [inline public/resources/js/stats.js:944]: I don't see a max amount of favorites here, is it set somewhere else?

**Review by metalcupcake5 — CHANGES_REQUESTED** [review]
Besides my comment, everything else looks good.
I would prefer it to be a gold star when someone has been selected as a favorite, but that's just my preference

File: public/resources/js/stats.js (hunk 1 above)
Comment (metalcupcake5) [reply public/resources/js/stats.js:944]: sigh it didn't select the correct parts whatever

**Comment by MartinNemi03** [comment]
> 
> 
> Besides my comment, everything else looks good.
> I would prefer it to be a gold star when someone has been selected as a favorite, but that's just my preference

Yeah, that's your preference, we had a poll on Discord and this was ultimately desided.

File: public/resources/js/stats.js (hunk 1 above)
Comment (nstringham) [reply public/resources/js/stats.js:946]: thanks, I forgot to add that it should be there now

**Comment by Pottedplant530** [comment]
@metalcupcake5  are you ready to sign off on it?

**Review by MartinNemi03 — CHANGES_REQUESTED** [review]
maybe rearrange things in getExtra, so it doesn't check the db for uuids of favorites on every page load (that's why it checks the db for favorites only on index). we had many problems with that before, so i dont think we want to reintroduce this issue.

**Comment by nstringham** [comment]
@MartinNemi03 thanks for pointing that out it should be fixed in eee2796

**Review by metalcupcake5 — APPROVED** [review]

**Review by MartinNemi03 — APPROVED** [review]

**Pull request merged**